app/modules/munkiinfo/scripts/munkiinfo.py

```diff
@@ -96,6 +98,26 @@ def get_munkiprotocol():
     except AttributeError:
         return 'Could not obtain protocol'
 
+def middleware_checks():
+    """Check for middleware, get version if supported"""
+
+    middleware_version = []
+    name = []
+    for filename in os.listdir('/usr/local/munki'):
+            print "checking %s" % filename
+            if (filename.startswith('middleware') 
+                and os.path.splitext(filename)[1] == '.py'):
+                
+                name = os.path.splitext(filename)[0]
+                print 'middleware name %s' % name
+                try:
+                    middleware = importlib.import_module(name)
+                    middleware_version = middleware.get_version()
+                except:
+                    print "Error importing middleware for version checks, or version function does not exist."
```

Comment:
From what I've seen so far (and it hasn't been much because middleware isn't heavily used yet), Wade's code is the reference implementation but most admins who are using it seem to be rolling their own at least to some degree - hence the update to the readme flagging the need to provide the get_version() function. I'll chat with Wade separately and see if he has interest in PRs for this. It would be a good idea to also add it to the munki documentation if Wade is open to PRs